agents/agent.py

```diff
@@ -721,3 +721,130 @@ def maybe_add_to_queue(self, data):
 
         if command_history.command in self.whitelist_commands:
             self.queue.put(command_history)
+
+    def get_cpu_usage(self, interval=60):
+        """
+        Get the average CPU usage percentage over the last minute.
+        """
+        try:
+            return psutil.cpu_percent(interval=interval)
+        except Exception as e:
+            maybe_log_message(
+                'Error getting CPU usage: %s' % str(e),
+                logger=self.logger,
+                fallback_logger=self.fallback_logger,
+            )
+            return -1.0
+
+    def get_ram_usage(self):
+        """
+        Get the current RAM usage percentage.
+        """
+        try:
+            mem = psutil.virtual_memory()
+            return mem.percent
+        except Exception as e:
+            maybe_log_message(
+                'Error getting RAM usage: %s' % str(e),
+                logger=self.logger,
+                fallback_logger=self.fallback_logger,
+            )
+            return -1.0
+
+    def get_load_average(self):
+        """
+        Get the system load average over the last 1 minute.
+        """
+        try:
+            return os.getloadavg()[0]
+        except Exception as e:
+            maybe_log_message(
+                'Error getting load average: %s' % str(e),
+                logger=self.logger,
+                fallback_logger=self.fallback_logger,
+            )
+            return -1.0
+
+    def get_disk_usage(self):
+        """
+        Get the current disk usage percentage for the root filesystem.
+        """
+        try:
+            usage = psutil.disk_usage('/')
+            return usage.percent
+        except Exception as e:
+            maybe_log_message(
+                'Error getting disk usage: %s' % str(e),
+                logger=self.logger,
+                fallback_logger=self.fallback_logger,
+            )
+            return -1.0
+
+    def generate_report(self):
```

Comment:
Respect existing naming convention: we have `status_to_dict` - this method can be `metric_to_dict`.